nylas/client/restful_model_collection.py

```diff
@@ -22,19 +22,26 @@ def __iter__(self):
         return self.values()
 
     def values(self):
+        limit = self.filters.get("limit")
         offset = self.filters["offset"]
+        fetched = 0
         while True:
-            models = self._get_model_collection(offset, CHUNK_SIZE)
+            if limit:
+                if fetched >= limit:
+                    break
+
+                req_limit = min(CHUNK_SIZE, limit - fetched)
```

Comment:
looks like we want to have a hard limit of `CHUNK_SIZE` that we send to `_get_model_collection` at a time. So if we have a limit > `CHUNK_SIZE` we loop and grab models in batches of `CHUNK_SIZE` or whatever we need remaining until we've gotten `limit` number of models. 

I must be missing something - how are we breaking out of the loop if the limit isn't set? Will something cause `self._get_model_collection(offset + fetched, req_limit)` to return falsey the second time around?

Reply:
> So if we have a limit > `CHUNK_SIZE` we loop and grab models in batches of `CHUNK_SIZE` or whatever we need remaining until we've gotten `limit` number of models.

That's right. It's a fairly typical pagination system, I believe.

> how are we breaking out of the loop if the limit isn't set?

Yep, I wrestled with this for awhile while writing this pull request. The basic problem is that our API doesn't expose any information about pagination, so the client has to guess. If you send an API request without a `limit` specified, and the response contains 100 items, how can you tell if there are 100 items in total, or if there more items to fetch on the next page? You can't! The only way to know is to _ask_ for the next page (by repeating the API request with `offset=100`), and see if you get more items or not. That's how we break out of the loop if the limit isn't set -- you'll notice that a few lines down, there's a check for `if not models: break`. If we make an API request that returns nothing, then break out of the loop.

Clients could avoid this pointless API request if we exposed pagination information, [like how GitHub does in their API](https://developer.github.com/v3/guides/traversing-with-pagination/). It would be nice if we could be explicit about when the client has reached the last page of data, rather than forcing clients to keep requesting until there's no more data to send. But that would require a change to the API, which is an entirely different codebase.

Reply:
Ah I see.  It would indeed be nice to return that info!
This makes sense to me now, but what do you think about adding a comment to clarify this behavior to someone coming in with less context?